src/main/java/com/sap/sgs/phosphor/fosstars/data/github/OwaspSecurityLibraries.java

```diff
@@ -56,13 +65,45 @@ protected ValueSet fetchValuesFor(GitHubProject project) throws IOException {
     logger.info("Figuring out if the project uses OWASP security libraries ...");
 
     LocalRepository repository = GitHubDataFetcher.localRepositoryFor(project);
+
     ValueSet values = new ValueHashSet();
+
     checkMaven(repository, values);
+    if (found(values)) {
+      return values;
+    }
+
     checkGradle(repository, values);
+    if (found(values)) {
+      return values;
+    }
+
+    values.update(USES_OWASP_ESAPI.value(false));
+    values.update(USES_OWASP_JAVA_ENCODER.value(false));
+    values.update(USES_OWASP_JAVA_HTML_SANITIZER.value(false));
 
     return values;
   }
 
+  /**
+   * Checks if a value set contains the features for OWASP security tools.
+   *
+   * @param values The value set to be checked.
+   * @return True if the value set contains the features, false otherwise.
+   */
+  private static boolean found(ValueSet values) {
+    return values.has(USES_OWASP_ESAPI)
+        && values.has(USES_OWASP_JAVA_ENCODER)
```

Comment:
Should this be connected via `&&`. Does it mean that all the features are inter-dependent?

Reply:
Yes, the method check if all the features are set.

> Does it mean that all the features are inter-dependent?

I am not sure if I got it. The features are independent.

Reply:
Is there a possibility that only one of the features may contain a value. But I guess not, all the features will be assigned a default value.

Reply:
The provider should assign a value to all features. If one of the features don't have a value after checking Maven files, then the provide tries to read Gradle files. I've noticed that `checkMaven()` method always sets all values, therefore `checkGradle()` can't be reached because `found()` just uses `ValueSet.has()`. Maybe you meant this problem. That's definitely an issue. I've fixed that. Thanks for paying attention @sourabhsparkala !

Reply:
Yes, I meant somewhere along the lines. But not exactly as clear as you have explained. This may be a bug in `OwaspDependencyScan` DataProvider as well?

Reply:
Yes, we can consider updating this provider. Could you please open an issue?

Reply:
https://github.com/SAP/fosstars-rating-core/issues/323